### ISSUE TITLE
Restart needed to activate daemon flags

### ISSUE THREAD

**Issue by amarandon** [nickjj/ansible-docker#42]
Hi Nick, I hope you're well.

I have an issue with the new `docker__daemon_flags`. Here is a playbook:
```yaml
---

- hosts: all
  become: true
  tasks:
    - name: Install required packages
      apt:
        state: present
        update_cache: true
        name:
          - python-pip
          - python-setuptools

- hosts: all
  become: true
  vars:
    docker__daemon_flags:
      - "-H unix://"
      - "-H tcp://0.0.0.0:5432"
  roles:
    - role: nickjj.docker
```
When I run this on a fresh Debian 9, I find that dockerd has been started without the extra flag:

```console
debian@playground:~$ ps ax|grep dockerd                                                                                                                                                                           
 4080 ?        Ssl    0:00 /usr/bin/dockerd -H unix://                                                                                                                                                            
 4414 pts/0    S+     0:00 grep dockerd
```
However the flag has been written properly to the systemd file:
```console
debian@playground:~$ cat /etc/systemd/system/docker.service.d/options.conf                                                                                                                                        
# Ansible managed                                                                                                                                                                                                 
                                                                                                                                                                                                                  
[Service]                                                                                                                                                                                                         
ExecStart=                                                                                                                                                                                                        
ExecStart=/usr/bin/dockerd -H unix:// -H tcp://0.0.0.0:5432
```
So I restart and this time dockerd does get started with the extra flag:
```console
debian@playground:~$ sudo systemctl restart docker.service
debian@playground:~$ ps ax|grep dockerd
 4420 ?        Ssl    0:00 /usr/bin/dockerd -H unix:// -H tcp://0.0.0.0:5432
 4537 pts/0    S+     0:00 grep dockerd
```

**Comment by nickjj** [comment]
Thanks for the heads up.

Looks like both daemon configuration tasks need `notify: ["Restart Docker"]` attached to them.

I will try to get to that today.

**Comment by amarandon** [comment]
Adding `notify: ["Restart Docker"]` to `Configure Docker daemon options (flags)` solves the issue.

However isn't it what `Reload systemd daemon` is supposed to do when flags are changed?

**Comment by nickjj** [comment]
That reload daemon command is only for when the systemd unit file itself changes.

**Comment by nickjj** [comment]
This has been fixed. It's available in v1.6.0.
